src/desktop.vala

```diff
@@ -226,25 +223,40 @@ class Desktop: Profile
 
   void show_settings (string panel)
   {
-    if (Environment.get_variable ("MIR_SOCKET") != null)
-      UrlDispatch.send ("settings:///system/bluetooth");
-    else if (is_desktop ("Unity") && Environment.find_program_in_path ("unity-control-center") != null)
+
+#if HAS_URLDISPATCHER
+
+        if (Environment.get_variable ("MIR_SOCKET") != null)
+        {
+            UrlDispatch.send ("settings:///system/bluetooth");
+
+            return;
+        }
+
+#endif
+
+    if (is_desktop ("Unity") && Environment.find_program_in_path ("unity-control-center") != null)
+    {
       spawn_command_line_async ("unity-control-center " + panel);
+    }
+    else if (is_desktop ("MATE") && Environment.find_program_in_path ("blueman-manager") != null)
+    {
+      spawn_command_line_async ("blueman-manager");
+    }
     else
+    {
       spawn_command_line_async ("gnome-control-center " + panel);
+    }
   }
 
   Action create_discoverable_action (Bluetooth bluetooth)
   {
     var action = new SimpleAction.stateful ("desktop-discoverable",
-                                            null,
+                                            VariantType.BOOLEAN,
                                             new Variant.boolean (bluetooth.discoverable));
 
-    action.activate.connect (()
-        => action.set_state (new Variant.boolean (!action.get_state().get_boolean())));
-
-    action.notify["state"].connect (()
-        => bluetooth.try_set_discoverable (action.get_state().get_boolean()));
+    action.activate.connect ((action, state)
```

Comment:
Same here 

Why did action.activate.connect get replaced with action.change_state.connect functions and action.change_state.connect gets completely removed

Reply:
> Same here
> 
> Why did action.activate.connect get replaced with action.change_state.connect functions and action.change_state.connect gets completely removed

Please see https://github.com/AyatanaIndicators/ayatana-indicator-bluetooth/pull/1#discussion_r496245536